test/testsizeof.cpp

```diff
@@ -162,6 +163,76 @@ class TestSizeof : public TestFixture {
                       "[test.cpp:5]: (warning, inconclusive) Found calculation inside sizeof().\n", errout.str());
     }
 
+    void sizeofFunction() {
+        check("class Foo\n"
+              "{\n"
+              "    int bar() { return 1; };\n"
+              "}\n"
+              "Foo f;int a,sizeof(f.bar())");
+        ASSERT_EQUALS("[test.cpp:5]: (warning) Found function call inside sizeof().\n", errout.str());
+
+        check("class Foo\n"
+              "{\n"
+              "    int bar() { return 1; };\n"
+              "    int bar() const { return 1; };\n"
+              "}\n"
+              "Foo f;int a,sizeof(f.bar())");
+        ASSERT_EQUALS("", errout.str());
+
+        check("class Foo\n"
+              "{\n"
+              "    int bar() { return 1; };\n"
+              "}\n"
+              "Foo * fp;int a,sizeof(fp->bar())");
+        ASSERT_EQUALS("[test.cpp:5]: (warning) Found function call inside sizeof().\n", errout.str());
+
+        check("#define foo() int\n"
+              "int a,sizeof(foo())");
+        ASSERT_EQUALS("", errout.str());
+
+        check("int foo() { return 1; }; int a,sizeof(foo())");
+        ASSERT_EQUALS("[test.cpp:1]: (warning) Found function call inside sizeof().\n", errout.str());
+
+        check("#define M(x) sizeof(x)\n"
+              "int foo() { return 1; }; int a,M(foo())");
+        ASSERT_EQUALS("", errout.str());
```

Comment:
this is not properly tested because we don't run the preprocessor in the tests. you can write a test code and indicate that certain tokens are expanded macro(s) using the special character $.

    int a = $sizeof $( foo() $) ;

that would mean that the `Token::isExpandedMacro()` would be true for the outer "sizeof (" and ")" tokens. And it would be false for `foo()`.